socketio_block.py

```diff
@@ -238,12 +243,20 @@ def handle_data(self, data):
 
     def _connect_to_socket(self):
         try:
+            # Don't need the job any more
+            self._connection_job = None
+
             self._do_handshake()
 
             url = self._get_ws_url()
 
             self._logger.debug("Connecting to %s" % url)
 
+            # In case the client is sticking around, close it before creating a
+            # new one
+            if self._client:
+                self._client.close_connection()
```

Comment:
Ah, I see, on the WebSocket. It's odd that WebSocketBaseClient.close
doesn't call that method...I kinda like close_connection because it
actually closes the socket. Maybe it should be both?

On Mon, Jan 5, 2015 at 11:20 AM, Matt Dodge <notifications@github.com>
wrote:

> In socketio_block.py
> <https://github.com/nio-blocks/socketio/pull/7#discussion-diff-22482592>:
>
> >              self._do_handshake()
> >
> >              url = self._get_ws_url()
> >
> >              self._logger.debug("Connecting to %s" % url)
> >
> > +            # In case the client is sticking around, close it before creating a
> > +            # new one
> > +            if self._client:
> > +                self._client.close_connection()
>
> I got it from here:
>
> https://github.com/Lawouach/WebSocket-for-Python/blob/master/ws4py/websocket.py#L200
>
> But you may be right, close may be preferable since it is used in the
> docs everywhere
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nio-blocks/socketio/pull/7/files#r22482592>.
>